var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -162,3 +162,42 @@ def cmake_args(self):
                 args.append("-DCMAKE_CUDA_FLAGS=-arch sm_{0}".format(cuda_arch[0]))
 
         return args
+
+    def test_smoke_test(self, source_dir=None):
+        """Build and run ctests"""
+        if source_dir is None:
+            source_dir = self.test_suite.current_test_cache_dir
+        spec = self.spec
+        with working_dir(source_dir):
+            with open("CMakeLists.txt", "w") as f:
+                f.write(
+                    """
+                        cmake_minimum_required(VERSION 3.12...3.21 FATAL_ERROR)
```

Comment:
Did the min version number get bumped here?  zfp 1.0.1 is still on 3.9 (to be bumped to 3.10 soon as 3.9 is being phased out), with 3.21 needed only for HIP (CUDA/HIP on `staging` require 3.23).  Are there CMake features being used here that require 3.12?

Reply:
good catch, not really, find this resolved at the last push

Reply:
Oh I just recalled why we needed this. since 3.12 this policy is introduced: https://cmake.org/cmake/help/latest/policy/CMP0074.html . We need it for cmake to find ZFP by using `ZFP_ROOT`.

Reply:
I see.  Should we wait for another commit to revert to 3.12, then?  The latest I'm seeing references both 3.9 and `ZFP_ROOT`.